File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        d={}
        for i in range(0,len(nums)):
            diff = target - nums[i]
            if diff in d:
                return [i,d[diff]]
            d[nums[i]]=i

        return []
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> d = new HashMap<>();

        for(int i=0; i < nums.length; i++){
            int diff = target - nums[i];

            if(d.containsKey(diff)){
                return new int[] {d.get(diff), i};
            }

            d.put(nums[i], i);
        }
        return new int[] {};
    }
}